src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -221,8 +221,7 @@ private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy,
      */
     private static boolean shouldBeAloneOnLineWithAloneOption(Details details,
                                                               String targetSrcLine) {
-        return !isAloneOnLine(details, targetSrcLine)
-                && !isEmptyBody(details.lcurly);
+        return !isAloneOnLine(details, targetSrcLine);
```

Comment:
I have tracked this `isEmptyBody` line back to https://github.com/checkstyle/checkstyle/commit/f1dced161ab0c818f0e1e125fc5b88b71f5b171e#diff-9b2a5f9e69b254b923cb4fd7420cb9a4R187
````
         else if (getAbstractOption() == RightCurlyOption.ALONE
-                && rcurly.getLineNo() == nextToken.getLineNo()
+                && !isAloneOnLine(details)
+                && !isEmptyBody(lcurly)) {
````

@romani It seems this was specifically requested and allowed in https://github.com/checkstyle/checkstyle/issues/1019 .
Please review.

Reply:
However, I believe this line is more like a mistake. The goal of that pr is to add a new option `ALONE_OR_SINGLELINE`, but it modified the behavior of `ALONE`. So maybe it is indeed a mistake.

Reply:
This [comment](https://github.com/checkstyle/checkstyle/blame/5a65f636aa6b08321ac8900bb1214a1a0fd410d6/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyLeft.java#L144) is weird also.

Reply:
Your are correct @fzdy1914 . Looking at the issue more closely, it is just introducing that option and wasn't suppose to be changing `alone` from what I can see. Changes were noticed in the PR at https://github.com/checkstyle/checkstyle/pull/1259#issuecomment-116502766 but were deemed good and not fully related to this issue.

I don't see any reason why we shouldn't continue this PR.
@romani if you are ok with everything, mark this discussion as resolved.

Reply:
it was mistake to affect `alone` mode